test/e2e/e2e_suite_test.go

```diff
@@ -0,0 +1,192 @@
+
+	vllmSimTag        = env.GetEnvString("VLLM_SIMULATOR_TAG", "dev", ginkgo.GinkgoLogr)
+	routingSideCarTag = env.GetEnvString("ROUTING_SIDECAR_TAG", "v0.2.0", ginkgo.GinkgoLogr)
+
+	existsTimeout     = getTimeout("EXISTS_TIMEOUT", defaultExistsTimeout)
+	readyTimeout      = getTimeout("READY_TIMEOUT", defaultReadyTimeout)
+	modelReadyTimeout = getTimeout("MODEL_READY_TIMEOUT", defaultModelReadyTimeout)
+	interval          = defaultInterval
+)
+
+func TestEndToEnd(t *testing.T) {
+	gomega.RegisterFailHandler(ginkgo.Fail)
+	ginkgo.RunSpecs(t,
+		"End To End Test Suite",
+	)
+}
+
+var _ = ginkgo.BeforeSuite(func() {
+	port = "30080"
+
+	setupK8sCluster()
+	setupK8sClient()
+	createCRDs()
+	createEnvoy()
+	applyYAMLFile(rbacManifest)
+	applyYAMLFile(serviceAccountManifest)
+	applyYAMLFile(servicesManifest)
+
+	infPoolYaml := readYaml(inferExtManifest)
+	infPoolYaml = substituteMany(infPoolYaml, map[string]string{"${POOL_NAME}": modelName + "-inference-pool"})
+	createObjsFromYaml(infPoolYaml)
+})
+
+var _ = ginkgo.AfterSuite(func() {
+	command := exec.Command("kind", "delete", "cluster", "--name", "e2e-tests")
+	session, err := gexec.Start(command, ginkgo.GinkgoWriter, ginkgo.GinkgoWriter)
+	gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+	gomega.Eventually(session).WithTimeout(600 * time.Second).Should(gexec.Exit(0))
+})
+
+// Create the Kubernetes cluster for the E2E tests and load the local images
+func setupK8sCluster() {
+	command := exec.Command("kind", "create", "cluster", "--name", "e2e-tests", "--config", "-")
+	stdin, err := command.StdinPipe()
+	gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+	go func() {
+		defer func() {
+			err := stdin.Close()
+			gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+		}()
+		clusterConfig := strings.ReplaceAll(kindClusterConfig, "${PORT}", port)
+		_, err := io.WriteString(stdin, clusterConfig)
+		gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+	}()
+	session, err := gexec.Start(command, ginkgo.GinkgoWriter, ginkgo.GinkgoWriter)
+	gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+	gomega.Eventually(session).WithTimeout(600 * time.Second).Should(gexec.Exit(0))
+
+	command = exec.Command("kind", "--name", "e2e-tests", "load", "docker-image",
+		"ghcr.io/llm-d/llm-d-inference-sim:"+vllmSimTag)
+	session, err = gexec.Start(command, ginkgo.GinkgoWriter, ginkgo.GinkgoWriter)
+	gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+	gomega.Eventually(session).WithTimeout(600 * time.Second).Should(gexec.Exit(0))
+
+	command = exec.Command("kind", "--name", "e2e-tests", "load", "docker-image",
+		"ghcr.io/llm-d/llm-d-inference-scheduler:dev")
+	session, err = gexec.Start(command, ginkgo.GinkgoWriter, ginkgo.GinkgoWriter)
+	gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+	gomega.Eventually(session).WithTimeout(600 * time.Second).Should(gexec.Exit(0))
+
+	command = exec.Command("kind", "--name", "e2e-tests", "load", "docker-image",
+		"ghcr.io/llm-d/llm-d-routing-sidecar:v0.2.0")
```

Comment:
Please consider using TAG overrides for all images, not just the simulator as in L122, but also for inference-scheduler (L128) and sidecar (L134).
If this does not make sense, feel free to disregard, just please ack since I do not see this addressed from the previous comments.

Reply:
Tags can now be specified in environment variables for all of the images.